lib/fountain-outline-view.coffee

```diff
@@ -1,340 +1,343 @@
+        @setActiveEditorBuffer(newFileText)
+      else
+        # update view manually since nothing changed
+        @updateList()
+
+    sortable = Sortable.create(outlineElement, {
+      onChoose: @onChoose
+      onUpdate: @onUpdate
+    })
+    sortable
+
+  flatten: (structuredScenes, flattenedRep) ->
+    i = 0
+    while i < structuredScenes.length
+      currentScene = structuredScenes[i]
+      flattenedRep.push currentScene
+      if currentScene.hasOwnProperty('children') and currentScene.children.length > 0
+        @flatten(currentScene.children, flattenedRep)
+      i++
+    flattenedRep
+
+  getOldLineIndexes: (oldFileLines, sceneList, movingElement) =>
+    # grab details about scene before array mutates
+    oldStartLine = parseInt(movingElement.item.attributes[1].value)
+    oldEndLine = parseInt(movingElement.item.attributes[2].value)
+#    console.debug("getOldIndexLines", oldStartLine, oldEndLine)
+    [oldStartLine, oldEndLine]
+
+  getNewStartLineIndex: (oldFileLines, sceneList, oldIndex, newIndex) =>
+    newStartLine = null
+
+    # account for array index changes
+    if (newIndex > oldIndex)
+      newIndex += 1
+
+    # if not the last array position
+    if (sceneList[newIndex])
+
+      if (@scenesHidden)
+
+        targetIndexIsScene = sceneList[newIndex].type == 'scene'
+        targetIsFirstChild = !sceneList[newIndex-1] || (sceneList[newIndex-1].endline == sceneList[newIndex].parentEndline)
+
+        if (!targetIndexIsScene && targetIsFirstChild)
+          newStartLine = parseInt(sceneList[newIndex].line)
+        else
+          # index to the parent to place after all children
+          newStartLine = parseInt(sceneList[newIndex-1].endline)
+
+      else
+        newStartLine = parseInt(sceneList[newIndex].line)
+
+    else
+
+      # if has preceding sibling
+      if (sceneList[newIndex - 1])
+        newStartLine = sceneList[newIndex - 1].endline
+      else
+        # they can manage any newline gaps
+        newStartLine = oldFileLines.length - 1
+
+    # yea, this is an intermediate value
+    # because we need slice lengths
+    newStartLine
+
+  getNewFileText: (oldFileLines, oldStartLine, oldEndLine, newStartLine) =>
+
+    # isolate relevant chunks
+    newFileText = ''
+    movingText = oldFileLines.slice(oldStartLine, oldEndLine)
+    textBefore = oldFileLines.slice(0, oldStartLine)
+    textAfter = oldFileLines.slice(oldEndLine)
+
+    # determine placement of text in preceding or trailing slice
+    if newStartLine < textBefore.length
+      textBefore.splice.apply(textBefore, [newStartLine, 0].concat(movingText))
+    else
+      textAfter.splice.apply(textAfter, [newStartLine - textBefore.length - movingText.length, 0].concat(movingText))
+
+    newFileText = textBefore.concat(textAfter).join('\n')
+    newFileText
+
+  setActiveEditorBuffer: (newFileText) =>
+    if editor = atom.workspace.getActiveTextEditor()
```

Comment:
I'm wondering if this might be picking up the new dock... or something else.  That's one possible explanation.